tensorflow/core/kernels/data/parallel_interleave_dataset_op.cc

```diff
@@ -52,592 +91,542 @@ constexpr char kDatasetName[] = "ParallelInterleaveV2";
+                      stats_aggregator->AddScalar(
+                          stats_utils::ThreadUtilizationScalarName(
+                              dataset()->node_name()),
+                          static_cast<float>(current_num_calls_) /
+                              static_cast<float>(num_parallel_calls_->value),
+                          num_elements());
+                    }
+                  }));
+            }
           }
-          cond_var_->notify_all();
         }
-      }
-
-      void EnsureThreadsStarted(IteratorContext* ctx)
-          EXCLUSIVE_LOCKS_REQUIRED(*mu_) {
-        if (!current_elements_manager_) {
-          auto new_ctx = std::make_shared<IteratorContext>(*ctx);
-          current_elements_manager_ = ctx->StartThread(
-              "tf_data_parallel_interleave_current",
-              [this, new_ctx]() { CurrentElementsManager(new_ctx); });
-        }
-        if (!future_elements_manager_) {
-          auto new_ctx = std::make_shared<IteratorContext>(*ctx);
-          future_elements_manager_ = ctx->StartThread(
-              "tf_data_parallel_interleave_future",
-              [this, new_ctx]() { FutureElementsManager(new_ctx); });
+        const auto& stats_aggregator = ctx->stats_aggregator();
+        if (stats_aggregator) {
+          stats_aggregator->AddScalar(
+              stats_utils::ThreadUtilizationScalarName(dataset()->node_name()),
+              static_cast<float>(current_num_calls_) /
+                  static_cast<float>(num_parallel_calls_->value),
+              num_elements());
         }
+        cond_var_->notify_all();
       }
+    }
 
-      // Fetches up to `dataset()->block_length_` results from `element`.
-      void FetchResults(const std::shared_ptr<IteratorContext>& ctx,
-                        const std::shared_ptr<Element>& element,
-                        int64 num_results, std::function<void()> done)
-          LOCKS_EXCLUDED(*mu_) {
-        RecordStart(ctx.get());
-        auto cleanup = gtl::MakeCleanup([this, ctx] { RecordStop(ctx.get()); });
-        bool end_of_input = false;
-        for (int64 i = 0; i < num_results; ++i) {
-          auto result = std::make_shared<Result>();
-          result->status = element->iterator->GetNext(
-              ctx.get(), &result->return_values, &end_of_input);
-          if (end_of_input) {
-            break;
-          }
-          RecordBufferEnqueue(ctx.get(), result->return_values);
-          mutex_lock l(*mu_);
-          mutex_lock l2(element->mu);
-          element->results.push_back(result);
-          result->is_ready = true;
-          cond_var_->notify_all();
-        }
+    void EnsureThreadsStarted(IteratorContext* ctx)
+        EXCLUSIVE_LOCKS_REQUIRED(*mu_) {
+      if (!current_elements_manager_) {
+        auto new_ctx = std::make_shared<IteratorContext>(*ctx);
+        current_elements_manager_ = ctx->StartThread(
+            "tf_data_parallel_interleave_current",
```

Comment:
let's turns `"tf_data_parallel_interleave_current"` and `"tf_data_parallel_interleave_future"` into named constants